pkg/provider/pulumi/azure/azure.go

```diff
@@ -186,42 +193,63 @@ func (a *azureProvider) Validate() error {
 		errList.Push(utils.NewNotSupportedErr(fmt.Sprintf("region %s not supported on provider %s", a.sc.Region, a.sc.Provider)))
 	}
 
-	if a.sc.Org == "" {
-		errList.Push(fmt.Errorf("target %s requires \"org\"", a.sc.Provider))
+	def, dok := a.sc.Config[stack.DefaultFunctionConfig]
+
+	if !dok || (a.sc.Org == "" && def.Org == nil) {
+		errList.Push(fmt.Errorf("config.default %s requires \"org\"", a.sc.Provider))
 	}
 
-	if a.sc.AdminEmail == "" {
-		errList.Push(fmt.Errorf("target %s requires \"adminemail\"", a.sc.Provider))
+	if !dok || (a.sc.AdminEmail == "" && def.AdminEmail == nil) {
+		errList.Push(fmt.Errorf("config.default %s requires \"adminemail\"", a.sc.Provider))
 	}
 
 	for fn, fc := range a.sc.Config {
 		if fc.Memory != nil && *fc.Memory < 128 {
-			errList.Push(fmt.Errorf("function config %s requires \"memory\" to be greater than 128 Mi", fn))
+			errList.Push(fmt.Errorf("config.\"%s\" requires \"memory\" to be greater than 128 Mi", fn))
 		}
 
 		if fc.Timeout != nil && *fc.Timeout < 15 {
-			errList.Push(fmt.Errorf("function config %s requires \"timeout\" to be greater than 15 seconds", fn))
+			errList.Push(fmt.Errorf("config.\"%s\" requires \"timeout\" to be greater than 15 seconds", fn))
+		}
+
+		if fc.AdminEmail != nil && fn != stack.DefaultFunctionConfig {
+			errList.Push(fmt.Errorf("config.\"%s\".adminemail should be set as default.adminemail", fn))
+		}
+
+		if fc.Org != nil && fn != stack.DefaultFunctionConfig {
+			errList.Push(fmt.Errorf("config.\"%s\".org should be set as a default.org", fn))
 		}
 	}
 
 	return errList.Err()
 }
 
 func (a *azureProvider) Configure(ctx context.Context, autoStack *auto.Stack) error {
-	dc, dok := a.sc.Config["default"]
+	dc, dok := a.sc.Config[stack.DefaultFunctionConfig]
+	if !dok {
+		a.sc.Config[stack.DefaultFunctionConfig] = azureFunctionConfig{}
+	}
+
+	if dc.AdminEmail == nil {
+		dc.AdminEmail = &a.sc.AdminEmail
+	}
+
+	if dc.Org == nil {
+		dc.Org = &a.sc.Org
+	}
+
+	a.sc.Config[stack.DefaultFunctionConfig] = dc
```

Comment:
@asalkeld just want to confirm if this is a breaking change for existing stack configurations?